src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/UpdateRunner.cs

```diff
@@ -0,0 +1,173 @@
+    {
+        public static List<XDocument>? ConfigUpdate(ConfigUpdater updater, ILogger logger)
+        {
+            try
+            {
+                logger.LogDebug("Start updating configuration files...");
+                XDocument wcfConfig = updater.GenerateNewConfig();
+                XDocument appConfig = updater.UpdateOldConfig();
+                var result = new List<XDocument>
+                            {
+                                appConfig,
+                                wcfConfig
+                            };
+                logger.LogInformation("Finish updating configuration files.");
+                return result;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, "Error updating config files");
+                return null;
+            }
+        }
+
+        public static void WriteConfigUpdate(XDocument appConfig, XDocument wcfConfig, string path, ILogger logger)
+        {
+            try
+            {
+                logger.LogDebug("Start writing changes to configuration files...");
+                wcfConfig.Save(Path.Combine(Path.GetDirectoryName(path), "wcf.config"));
+                appConfig.Save(path);
+                logger.LogInformation("Finish writing changes to configuration files.");
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error writing changes to {path}");
+            }
+        }
+
+        public static XDocument? PackageUpdate(PackageUpdater packageUpdater, ILogger logger)
+        {
+            try
+            {
+                logger.LogDebug("Start updating project file...");
+                packageUpdater.UpdatePackages();
+                var projFile = packageUpdater.UpdateSDK();
+                logger.LogInformation("Finish updating project file.");
+                return projFile;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, "Error updating project file");
+                return null;
+            }
+        }
+
+        public static void WritePackageUpdate(XDocument projFile, string path, ILogger logger)
+        {
+            try
+            {
+                logger.LogDebug("Start writing changes to project file...");
+                projFile.Save(path);
+                logger.LogInformation("Finish writing changes to project file.");
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error writing changes to {path}");
+            }
+        }
+
+        public static SyntaxNode? SourceCodeUpdate(SourceCodeUpdater sourceCodeUpdater, ILogger logger)
+        {
+            try
+            {
+                logger.LogDebug("Start updating source code...");
+                var root = sourceCodeUpdater.UpdateDirectives();
+                root = sourceCodeUpdater.RemoveOldCode(sourceCodeUpdater.AddTemplateCode(root));
```

Comment:
As mentioned above, I think these calls (RemoveOldCode and AddTemplateCode - and, really, even UpdateDirectives) are likely to always happen together so it may make sense to have a single source code updater method that wraps them all.